pkg/operation/botanist/resource_manager.go

```diff
@@ -66,3 +68,35 @@ func (b *Botanist) WaitUntilManagedResourcesDeleted(ctx context.Context) error {
 		return retry.MinorError(fmt.Errorf("not all managed resources have been deleted in the shoot cluster (still existing: %s)", names))
 	})
 }
+
+// CleanManagedResourcesFinalizers cleans all managed resources finalizers.
+func (b *Botanist) CleanManagedResourcesFinalizers(ctx context.Context) error {
+	managedResources, err := b.listManagedResources(ctx)
```

Comment:
Why is this a separate function? I don't see it used anywhere else. Move its body here?